app/models/entities/opportunity.rb

```diff
@@ -0,0 +1,60 @@
+class Entities::Opportunity < Maestrano::Connector::Rails::Entity
+  def self.connec_entity_name
+    'Opportunity'
+  end
+
+  def self.external_entity_name
+    'Deal'
+  end
+
+  def self.mapper_class
+    OpportunityMapper
+  end
+
+  def self.object_name_from_connec_entity_hash(entity)
+    entity['name']
+  end
+
+  def self.object_name_from_external_entity_hash(entity)
+    entity['name']
+  end
+
+  def self.references
+    %w(lead_id)
+  end
+
+  def before_sync(last_synchronization_date = nil)
+     @stages = @external_client.get_entities('stages')
+     @stages ||= []
+  end
+
+  def map_to_connec(entity)
+    mapped_entity = super
+    stage = @stages.find { |stage| stage['id'] == entity['stage_id'] }
+    mapped_entity[:sales_stage] = stage['name'] ? stage['name'] : "stage not found"
```

Comment:
```ruby
if stage
mapped_entity[:sales_sage] = stage['name']
mapped_entity[:probability] = stage['likelihood']
end
mapped_entity
end
```

As the value are not mandatory in connec, I'd rather have no value rather than default random ones